manuscripts2/Readme.md

```diff
@@ -0,0 +1,230 @@
+# Readme
+
+This folder is an initialization to the new functions and classes that have to be added to manuscripts. This file provides a brief introduction to these classes and functions.
+
+
+### TODO:
+- Is popping from dict and then creating the nested aggregation the most efficient way to do this?
+- How do you parse 3 level nested aggregation? 
+
+
+### Structure:
+The main Class which is being implemented is the `EQCC` class. `EQCC`  stands for `Elasticsearch Query, Connection and Compute` as it provides a connection to elasticsearch, queries it and computes the response.
+
+The important variables inside the `EQCC` objects are as follows:
+
+- `s`: This is the elasticsearch_dsl Search object which is responsible of quering elasticsearch and getting the results. All the aggregations, queries and filters are applied to this Search object and then it queries elasticsearch and fetches the required results.
+- `parent_id`: This is a counter counting the aggregations that are applied to the Search object. It starts with `0` and increments as aggregations are added.
+- `queries`: This is a dictionary containing two lists: `must` and `must_not`. This dictionary contains the normal(`must`) and inverse(`must_not`) queries.
+- `aggregations`: This is an OrderedDict which contains all the aggregations applied. An ordered dict allows us to create nested aggregations easily, as we'll see below.
+- `child_agg_counter_dict`: This dict keeps a track of the number of child aggregations that have been applied in an aggregation.
```

Comment:
@jgbarah It is actually a dict which contains individual counters for all the nested aggregations that the object contains. Example:
```
prs = PullRequests(index)
prs.get_cardinality("id_in_repo").by_authors()
prs.get_sum("time_open_days").by_authors()
prs.get_percentile("time_to_close_days").by_authors()
```
here, the child_agg_counter_dict will contain one key `terms_author_uuid` (because `by_authors` is a `terms` aggregation and uses the field `author_uuid`) and it's value will be 3.
This aggregation will give us the number of prs, the sum of time_open_days of the open prs and the percentile of time_to_close_days of the closed PRs, all seggregated by authors, in one go!
This is a fuctionality that I think will decrease the amount of times we have to query elasticsearch and get the results.